src/Minimap.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const MINIMAP_WIDTH = 350;
+const MINIMAP_HEIGHT = 250;
+
+const Minimap = ({ vertices, extremeX, extremeY, viewport, changeScrollHandler, style }) => {
+  const getScalingFactor = () => ({
+    x: MINIMAP_WIDTH / extremeX,
+    y: MINIMAP_HEIGHT / extremeY,
+  });
+
+  const getMinimapViewport = () => ({
+    top: viewport.yMin * scalingFactor.y,
+    left: viewport.xMin * scalingFactor.x,
+    width: Math.min(viewportWidth * scalingFactor.x, MINIMAP_WIDTH),
+    height: Math.min(viewportHeight * scalingFactor.y, MINIMAP_HEIGHT),
+  });
+
+  const getMinimapNodes = () => {
+    return vertices.map(vertex => ({
+      top: vertex.top * scalingFactor.y,
+      left: vertex.left * scalingFactor.x,
+      width: vertex.width * scalingFactor.x,
+      height: vertex.height * scalingFactor.y,
+    }));
+  };
+
+  const renderNodes = () => {
+    const minimapNodes = getMinimapNodes();
+    return minimapNodes.map(node => (
+      <div
+        key={node.id}
+        style={{
+          boxSizing: 'border-box',
+          position: 'absolute',
+          top: `${node.top}px`,
+          left: `${node.left}px`,
+          width: `${node.width}px`,
+          height: `${node.height}px`,
+          backgroundColor: 'gold',
+          border: '1px solid black',
+        }}
+      />
+    ));
+  };
+
+  const ref = useRef(null);
+  const scalingFactor = getScalingFactor();
+
+  const viewportWidth = viewport.xMax - viewport.xMin;
+  const viewportHeight = viewport.yMax - viewport.yMin;
+
+  const minimapViewport = getMinimapViewport();
+
+  useEffect(() => {
```

Comment:
no dependencies in this useEffect ?

Reply:
also we are adding an event listener to a div by accessing it's, we can just make a function(handleClick) and add it in onClick of the div, don't see a need to use useEffect here